go/enclave/enclave.go

```diff
@@ -1386,3 +1417,31 @@ func describeBSR(response *common.BlockSubmissionResponse) string {
 	}
 	return fmt.Sprintf("%s, %s", producedBatch, producedRollup)
 }
+
+// handlePlainTextError captures possible underlying SystemErrors errors or returns PlaintextError
+func (e *enclaveImpl) handlePlainTextError(err error) (*responses.EnclaveResponse, common.SystemError) {
+	if errors.Is(err, errutil.InternalError{}) {
+		return nil, e.internalError(err)
+	}
+
+	return responses.AsPlaintextError(err), nil
+}
+
+// handleEncryptedError captures possible underlying SystemErrors errors or returns EncryptedError
+func (e *enclaveImpl) handleEncryptedError(err error, encryptor responses.ViewingKeyEncryptor) (*responses.EnclaveResponse, common.SystemError) {
```

Comment:
I find this method a bit confusing. 
The reason is that, in my mind, when you get around to creating the EnclaveResponse containing the encrypted user error, you would have already checked for system errors.

I suggest removing this method and just doing `return responses.AsEncryptedError(err, encryptor), nil` in all the places where you call it